src/handlers.cpp

```diff
@@ -2529,59 +2522,171 @@ void PushProfileTask::on_get_primary_impus_success(CassandraStore::Operation* op
     SAS::Event event(this->trail(), SASEvent::CACHE_GET_ASSOC_DEF_IMPU_FAIL, 1);
     SAS::report_event(event);
     send_ppa(DIAMETER_REQ_FAILURE);
+    delete this;
     return;
   }
 
-  // We can only look for a match if an IMS Subscription was present on the PPR.
   if (_ims_sub_present)
   {
-    // Attempt to find a match between any of the default ids of the sets the
-    // private id corresponds to, and the default id in the PPR. If no match is
-    // found the PPR is changing the default id (which is not permitted), so
-    // reject it. Otherwise, continue.
-    std::string new_default_id;
-    XmlUtils::get_default_id(_ims_subscription, new_default_id);
-    bool impu_match_found = false;
-    for (std::string current_default_id : _default_impus)
+    if (default_ids_match())
```

Comment:
impus not ids surely? In general your functions names/variables use ids where I think they should use impus for accuracy.